File: ABC/351-400/390/c.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,n) for(int i=0;i<n;++i)

int main(){
    int h,w;
    cin >> h >> w;
    vector<string> s(h);
    rep(i,h) cin >> s[i];
    const int INF = 1001001001;
    int left=INF,right=-1,top=INF,down=-1;
    rep(i,h){
        rep(j,w){
            if(s[i][j] == '#'){
                left = min(left,j);
                right = max(right,j);
                top = min(top,i);
                down = max(down,i);
            }
        }
    }
    for(int i=top;i<=down;i++){
        for(int j=left;j<=right;j++){
            if(s[i][j] == '.'){
                cout << "No" << endl;
                return 0;
            }
        }
    }
    cout << "Yes" << endl;
    return 0;
}

File: ABC/351-400/390/c.py
def main():
    h,w = map(int,input().split())
    s = [input() for _ in range(h)]
    
    l = w
    r = 0
    t = h
    d = 0
    
    for i in range(h):
        for j in range(w):
            if s[i][j] == '#':
                l = min(l,j)
                r = max(r,j)
                t = min(t,i)
                d = max(d,i)
                
    for i in range(t,d+1):
        for j in range(l,r+1):
            if s[i][j] == '.':
                print("No")
                return
    
    print("Yes")

if __name__ == "__main__":
    main()